src/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_cython.pyx

```diff
@@ -284,15 +281,15 @@ cdef class PyDBFrame:
 #     def trace_exception(self, frame, event, arg):
     # ENDIF
         if event == 'exception':
-            should_stop, frame = self.should_stop_on_exception(frame, event, arg)
+            should_stop, frame = self._should_stop_on_exception(frame, event, arg)
 
             if should_stop:
-                if self.handle_exception(frame, event, arg, EXCEPTION_TYPE_HANDLED):
+                if self._handle_exception(frame, event, arg, EXCEPTION_TYPE_HANDLED):
                     return self.trace_dispatch
 
         elif event == 'return':
             exc_info = self.exc_info
-            if exc_info:
+            if exc_info and arg is None:
```

Comment:
This is the actual fix.

It's not perfect because if the user has a return in the same line as the except

i.e.:

```
try:
   ...
except: return
```

The debugger will still think that it's exiting due to an exception.

I couldn't think of a better workaround (it's something that probably should be fixed in CPython itself -- i.e.: https://bugs.python.org/issue34099 -- maybe I could take a look at doing this change in CPython myself in the future).